rasa/core/events/__init__.py

```diff
@@ -212,6 +213,7 @@ def __init__(
         intent: Optional[Dict] = None,
         entities: Optional[List[Dict]] = None,
         parse_data: Optional[Dict[Text, Any]] = None,
+        message: Optional[Message] = None,
```

Comment:
@evgeniiaraz Why do we actually need the `message` parameter? The text and (in case of `ActionExecuted` the `name` is the text) are enough to featurize the events later, aren't they?

Reply:
I mean for the action it might make sense to introduce a `text` parameter just to not lumping everything together

Reply:
The idea was to store the `message` object so that we don't have to re-featurizemthe text once the NLU pipeline is trained (NLU will add the features to `message` during training automatically).

Reply:
Re-featurize when? During Core training? Don't we featurize the entire training data just once before training core? When would we re-use it?

Reply:
Yes, during Core training. If you're thinking about [here](https://github.com/RasaHQ/rasa/blob/62e88f887ad8381b47721eaa2a11c765b6d9deb7/rasa/core/interpreter.py#L349-L362), we re-use the features as much as we can and find the same featurized `message` when an utterance happened to be not featurized during NLU training.
I think the reason why some `message`s  are not featurized is: when `TrainingData` is created, we de-duplicate the training examples (inside `TrainingData`). So,  those which are duplicates are not featurized in place.

Reply:
Ok, I get it 👍 But as NLU training is now separated again, we don't have access to the NLU training data in the first place. I think we should rather do something like a lookup table during featurization to avoid duplicate predictions.